packages/devices/src/index.ts

```diff
@@ -73,3 +74,53 @@ export const createCameras = () => {
     equals: equalDeviceLists
   });
 };
+
+/**
+ * Creates a reactive wrapper to get device acceleration
+ * @param includeGravity boolean. default value false
+ * @param interval number as ms. default value 100
+ * @returnValue Acceleration: Accessor<DeviceMotionEventAcceleration | undefined>
+ */
+export const createAccelerometer = (includeGravity: boolean = false, interval: number = 100) => {
+  const [acceleration, setAcceleration] = createSignal<DeviceMotionEventAcceleration>();
+  let throttled = false;
+
+  const accelerationEvent = (e: DeviceMotionEvent) => {
+    if (throttled) return;
+    throttled = true
+    setTimeout(() => { throttled = false }, interval)
+
+    const acceleration = includeGravity ? e.accelerationIncludingGravity : e.acceleration;
+    setAcceleration(acceleration ? acceleration : undefined);
+  }
+
+  window.addEventListener("devicemotion", accelerationEvent);
+  getOwner() && onCleanup(() => { window.removeEventListener("devicemotion", accelerationEvent) });
+  return acceleration;
+};
+
+/**
+ * Creates a reactive wrapper to get device orientation
+ * @param interval number as ms. default value 100
+ * @returnValue { alpha: 0, beta: 0, gamma: 0 }
+ */
+export const createGyroscope = (interval: number = 100) => {
+  const [orientation, setOrientation] = createStore({ alpha: 0, beta: 0, gamma: 0 });
+  let throttled = false;
+
+  const orientationEvent = (e: DeviceOrientationEvent) => {
+    if (throttled) return;
+    throttled = true
+    setTimeout(() => { throttled = false }, interval)
```

Comment:
Shouldn't throttling be something you implement in a composable way, outside of the primitive?
If throttling is an important feature here, then maybe the user should be passing a scheduling callback to control the behavior. You can see this as an example of this pattern: https://github.com/solidjs-community/solid-primitives/tree/main/packages/bounds#throttling-updates

Reply:
Using a simpler solution is better for performance in this case.